### PR TITLE
_PATH_VARRUN can be set on configure

### PR DESCRIPTION
**Pull request by bsdsx**

### CODE REVIEW

**Comment by Vaelatern** [comment]
No, this would undo the work of the configure check. This file was well designed to begin with.

File: openbsd-compat/defines.h [hunk 1]
```diff
@@ -497,10 +497,6 @@ typedef uint16_t	in_port_t;
 #define FPARSELN_UNESCALL       0x0f
 #endif
 
-#ifndef _PATH_VARRUN
```
Comment (Vaelatern) [inline openbsd-compat/defines.h]: If defined, this doesn't get run. Thus when configure defines it. this doesn't get run. Exactly as it should be.

**Comment by poolpOrg** [comment]
huh nope nope nope. what are you achieving with this ?

File: configure.ac [hunk 2]
```diff
@@ -1200,7 +1200,7 @@ AC_SUBST([sockdir])
 AC_ARG_WITH([path-pidfile],
 	[  --with-path-pidfile=PATH	Specify path to smtpd.pid directory (default=/var/run)],
 	[AS_IF([test "x$with_path-pidfile" != "xno" -a "x$with_path-pidfile" != "xyes"],
-		AC_DEFINE_UNQUOTED([_PATH_VARRUN], ["$with_path-pidfile"])
+		AC_DEFINE_UNQUOTED([_PATH_VARRUN], ["$with_path-pidfile"], [Specify location of smtpd.pid])
```
Comment (Vaelatern) [inline configure.ac]: This change I can agree with, but doesn't seem critical.

**Comment by bsdsx** [comment]
on Solaris, there is no /usr/include/paths.h . _PATH_VARRUN is not defined. After ./configure, there is no _PATH_VARRUN defined anywhere

**Comment by bsdsx** [comment]
Without description, i get this error:
(CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/dsx/git/OpenSMTPD/missing autoheader)
autoheader: warning: missing template: _PATH_VARRUN
autoheader: Use AC_DEFINE([_PATH_VARRUN], [], [Description])
Makefile:393: recipe for target 'config.h.in' failed
gmake: *** [config.h.in] Error 1

**Comment by bsdsx** [comment]
./configure --with-cflags='-I /opt/csw/include -L /opt/csw/lib -D_XOPEN_SOURCE=600 -D__EXTENSIONS__=1 -D_XPG6 -D_PATH_DEFPATH=\"/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/local/bin\"' --with-path-empty=/var/opt/csw/empty --with-path-CAfile=/etc/opt/csw/ssl/cert.pem --prefix=/opt/csw --with-path-pidfile=/var/run/foobar_fail

Expected: _PATH_VARRUN define to '/var/run/foobar_fail' in config.h

grep -r foobar_fail
config.log:  $ ./configure --with-cflags=-I /opt/csw/include -L /opt/csw/lib -D_XOPEN_SOURCE=600 -D__EXTENSIONS__=1 -D_XPG6 -D_PATH_DEFPATH=\"/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/local/bin\" --with-path-empty=/var/opt/csw/empty --with-path-CAfile=/etc/opt/csw/ssl/cert.pem --prefix=/opt/csw --with-path-pidfile=/var/run/foobar_fail
config.status:ac_cs_config="'--with-cflags=-I /opt/csw/include -L /opt/csw/lib -D_XOPEN_SOURCE=600 -D__EXTENSIONS__=1 -D_XPG6 -D_PATH_DEFPATH=\\\"/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/local/bin\\\"' '--with-path-empty=/var/opt/csw/empty' '--with-path-CAfile=/etc/opt/csw/ssl/cert.pem' '--prefix=/opt/csw' '--with-path-pidfile=/var/run/foobar_fail'"
config.status:  set X /bin/bash './configure'  '--with-cflags=-I /opt/csw/include -L /opt/csw/lib -D_XOPEN_SOURCE=600 -D__EXTENSIONS__=1 -D_XPG6 -D_PATH_DEFPATH=\"/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/local/bin\"' '--with-path-empty=/var/opt/csw/empty' '--with-path-CAfile=/etc/opt/csw/ssl/cert.pem' '--prefix=/opt/csw' '--with-path-pidfile=/var/run/foobar_fail' $ac_configure_extra_args --no-create --no-recursion

No _PATH_VARRUN

**Comment by bsdsx** [comment]
Read my last comment:

https://github.com/OpenSMTPD/OpenSMTPD/issues/637#issuecomment-241246556

**Pull request closed without merging**